### PR TITLE
Implement resize_, resize_as_ for named tensors

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26543 Named tensor support for: atan2, output_nr, detach{_}, requires_grad_
* #26541 Support torch.pow with named tensors
* **#26493 Implement resize_, resize_as_ for named tensors**

resize_ and resize_as_ are low level functions that are not meant to be
used as a part of the regular PyTorch user's routine. However, they are
used to implement a lot of our operations: `out=` functionality is
implemented by resizing an output to be the correct size.

To keep in line with already implemented `out=` functionality, we do the
following:
- resize_as_(self, other) propagates names according to `out=` functionality.
This means that if self doesn't have names, then we propagate
other.names. If self does have names, they must be equal to other.names.

In addition, resize_ cannot resize a named tensor to anything but the same size.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17501404](https://our.internmc.facebook.com/intern/diff/D17501404)

### CODE REVIEW

File: aten/src/ATen/native/ResizeCommon.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#include <ATen/ATen.h>
+#include <ATen/NamedTensorUtils.h>
+
+namespace at { namespace native {
+
+#ifdef BUILD_NAMEDTENSOR
+inline Tensor& resize_named_tensor_(Tensor& self, IntArrayRef size) {
+  TORCH_INTERNAL_ASSERT(self.has_names());
+  TORCH_CHECK(
+      self.sizes() == size,
```
Comment (nairbv) [inline aten/src/ATen/native/ResizeCommon.h]: would it be possible to re-size from the right if left dimensions are wildcards? e.g. would it make sense for a `tensor([3,4,5], names=(*, *, 'foo')).resize(4,15) -> tensor([4,15], names=(*, 'foo'))` ?

**Review by nairbv — APPROVED** [review]

File: aten/src/ATen/native/ResizeCommon.h (hunk 1 above)
Comment (zou3519) [reply aten/src/ATen/native/ResizeCommon.h]: In your example, we're "mixing and matching" dimensions with other dimensions. In that case, it's unclear if the last dimension should retain the same name.

For example, someone can do `tensor([5], names=('foo',).resize(5, 1)` -> clearly the first dimension should now be the "foo" dimension and the dimension of size 1 is a brand new dimension.
